webapp/src/components/TodoList.vue

```diff
@@ -0,0 +1,215 @@
+        @delete-entry="deleteEntry"
+        @save-entry="saveEntry"
+        @remove-entry-local="removeEntryLocal"
+      />
+    </ol>
+  </div>
+</template>
+
+<script>
+import gql from "graphql-tag";
+import TodoItem from "./TodoItem";
+
+export default {
+  name: "TodoList",
+  components: { TodoItem },
+  props: ["user"],
+  apollo: {
+    todoListData: {
+      query: gql`
+        query {
+          todoListData: todos {
+            id
+            text
+            author {
+              name
+            }
+          }
+        }
+      `
+    }
+  },
+  data() {
+    return {
+      initialEditMode: false,
+      isEditingAuthor: false
+    };
+  },
+  methods: {
+    removeEntryLocal() {
+      this.todoListData.splice(this.todoListData.length - 1, 1);
+    },
+    addEntry() {
+      this.initialEditMode = true;
+      const newTodo = {
+        message: "New Todo",
+        author: {
+          name: this.user.username
+        }
+      };
+      this.todoListData.push(newTodo);
+    },
+    deleteEntry: function(todo) {
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation delToDo($id: ID) {
+              delToDo(id: $id) {
+                id
+                text
+                author {
+                  name
+                }
+              }
+            }
+          `,
+          variables: {
+            id: todo.id
+          }
+        })
+        .then(body => {
+          this.todoListData = body.data.delToDo.slice();
+        });
+    },
+    saveEntry: function(todo) {
+      if (todo.id == null) {
+        this.initialEditMode = false;
+        this.$apollo
+          .mutate({
+            mutation: gql`
+              mutation addToDo($text: String!, $authorName: String!) {
+                addToDo(text: $text, authorName: $authorName) {
+                  id
+                  text
+                  author {
+                    name
+                  }
+                }
+              }
+            `,
+            variables: {
+              text: todo.text,
+              authorName: this.user.username
+            }
+          })
+          .then(body => {
+            this.todoListData = body.data.addToDo.slice();
+          });
+      } else {
+        this.$apollo
+          .mutate({
+            mutation: gql`
+              mutation updateToDo(
+                $id: ID!
+                $text: String!
+                $authorName: String!
+              ) {
+                updateToDo(id: $id, text: $text, authorName: $authorName) {
+                  id
+                  text
+                  author {
+                    name
+                  }
+                }
+              }
+            `,
+            variables: {
+              id: todo.id,
+              text: todo.text,
+              authorName: this.user.username
+            }
+          })
+          .then(body => {
+            this.todoListData = body.data.updateToDo.slice();
```

Comment:
With `vue-apollo` it shouldn't be necessary to update all that data manually. If you make your `id` of type `ID` vue-apollo is able to update the already existing item in the cache.

Reply:
I don't know how to do this when manually calling the Query